lib/utils/widget.helper.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+// Get the state data of a widget at a specific position
+T? getTargetStateData<T>(BuildContext context, double x, double y) {
+  final renderBox = context.findRenderObject() as RenderBox;
+
+  final offset = renderBox.localToGlobal(Offset(x, y));
+
+  final HitTestResult result = HitTestResult();
+  WidgetsBinding.instance.hitTest(result, offset);
+
+  for (final i in result.path) {
+    if (i.target is RenderMetaData) {
+      final d = i.target as RenderMetaData;
+      if (d.metaData is T) {
+        return d.metaData as T;
+      }
+    }
+  }
+  return null;
+}

```

Comment:
helped from this : 
https://github.com/flutter/flutter/issues/19941#issuecomment-486950990

Reply:
that's great, but you should make it more understandable. currently, it's very hard @@

Reply:
IMO: Pls write ADR for this ticket